### PR TITLE
Removed semester and year hardcoding 

### PR DESCRIPTION
**Pull request by rstumbaugh**
Based off of when the University updates the SOC (found [here](https://docs.google.com/document/d/1ymAGvxoZ7EYoBmygv5RD1htDHp7rndqxBIWl5Okrrhw/edit)). Looks like historically it's been updated around mid-September for Spring classes and mid-February for Fall classes

### CODE REVIEW

**Comment by rstumbaugh** [comment]
The branch with `month <= 12` will only get reached if month < 3 or month > 10. If that's the case, then we want the spring semester's schedule, since that gets updated in September. So if the month is November or December, the Spring semester is in the next calendar year. Otherwise, it's Jan or Feb and we can use the current year. 

Also, I just thought 3 preceding years was decent. As of now I'm pretty sure Testudo doesn't even have data for 2014, so it looks like they only keep a year or two on the Schedule of Classes. I can write some tests to make sure that semesters `get_current_semester` is actually the current semester on the SOC and stuff like that.

File: app/helpers/courses_helpers.rb [hunk 1]
```diff
@@ -165,13 +165,15 @@ def time_to_int time
...
-          time.year.to_s + '08'
+        # Testudo schedules updated mid/late September for Spring, mid/late Feb for fall
+        # advisor calendar found here http://registrar.umd.edu/faculty-staff/
+        month = Time.now.month
+        year = Time.now.year
+        if month >= 3 && month <= 10
+          year.to_s + '08'
         else
-          (time.year + 1).to_s + '01'
+          (month <= 12 ? year + 1 : year).to_s + '01'
```
Comment (nickaversano) [inline app/helpers/courses_helpers.rb]: `month <= 12` is always true. no need for ternary

Comment (rstumbaugh) [inline app/helpers/courses_helpers.rb]: Oops, not too sure what I was thinking here.

File: app/helpers/courses_helpers.rb [hunk 2]
```diff
@@ -165,13 +165,15 @@ def time_to_int time
...
-          time.year.to_s + '08'
+        # Testudo schedules updated mid/late September for Spring, mid/late Feb for fall
+        # advisor calendar found here http://registrar.umd.edu/faculty-staff/
+        month = Time.now.month
+        year = Time.now.year
+        if month >= 3 && month <= 10
+          year.to_s + '08'
         else
-          (time.year + 1).to_s + '01'
+          ([11,12].include?(month) ? year + 1 : year).to_s + '01'
```
Comment (nickaversano) [inline app/helpers/courses_helpers.rb]: nit: prefer string interpolation to `.to_s`. e.g., `"#{year}08"`

File: app/helpers/courses_helpers.rb [hunk 3]
```diff
@@ -165,13 +165,15 @@ def time_to_int time
...
       def get_current_semester
-        time = Time.new
-        if time.month >= 3 && time.month < 10
-          time.year.to_s + '08'
+        # Testudo schedules updated mid/late September for Spring, mid/late Feb for fall
+        # advisor calendar found here http://registrar.umd.edu/faculty-staff/
+        month = Time.now.month
+        year = Time.now.year
+        if month >= 3 && month <= 10
+          year.to_s + '08'
```
Comment (nickaversano) [inline app/helpers/courses_helpers.rb]: same

File: app/helpers/courses_helpers.rb [hunk 4]
```diff
@@ -165,13 +165,15 @@ def time_to_int time
         time.to_i
       end
 
-      # TODO: make this line up with Testudo accurately and implement it in course controller
       def get_current_semester
```
Comment (nickaversano) [inline app/helpers/courses_helpers.rb]: nit: `current_semester` is probably nicer

**Comment by nickaversano** [comment]
awesome!

**Pull request merged**